tests/unit/test_charm.py

```diff
@@ -876,3 +876,43 @@ def test_wordpress_version_set(harness: ops.testing.Harness):
     harness.begin_with_initial_hooks()
 
     assert harness.get_workload_version() == WordpressContainerMock._WORDPRESS_VERSION
+
+
+@pytest.mark.usefixtures("attach_storage")
+def test_leader_installation_failure(
+    patch: WordpressPatch, harness: ops.testing.Harness, app_name
+):
+    """
+    arrange: charm peer and database relation is ready, the storage is attached.
+    act: start the charm as a follower unit.
+    assert: charm unit should enter blocked state, and the installation error should be seen
+        in the status.
```

Comment:
This isn't describing (to me at least) what the test is actually doing. Isn't the failure here because "endpoints" doesn't include a `:` and so `host` and `port` are set to `None`? All we're saying in `arrange` is "charm peer and database relation is ready, the storage is attached."

The name is `test_leader_installation_failure` but then we say here "start the charm as a follower unit" so that seems a bit inconsistent.